workflow/util/util_test.go

```diff
@@ -1048,3 +1050,280 @@ func TestGetTemplateFromNode(t *testing.T) {
 		assert.Equal(t, tc.expectedTemplateName, actual)
 	}
 }
+
+var retryWorkflowWithFailedNodeHasChildrenNodes = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  annotations:
+    workflows.argoproj.io/pod-name-format: v1
+  creationTimestamp: "2022-08-04T08:35:10Z"
+  generateName: test-pipeline-
+  generation: 10
+  labels:
+    workflows.argoproj.io/completed: "true"
+    workflows.argoproj.io/phase: Failed
+    workflows.argoproj.io/resubmitted-from-workflow: test-pipeline-t5h77
+  managedFields:
```

Comment:
Is this needed? Can we make this YAML more concise and only contain what's needed for the test?

Reply:
done